src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -371,6 +394,172 @@ class spell_pri_divine_hymn : public SpellScript
     }
 };
 
+// 392988 - Divine Image
+class spell_pri_divine_image : public AuraScript
+{
+    PrepareAuraScript(spell_pri_divine_image);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_DIVINE_IMAGE_SUMMON,
+            SPELL_PRIEST_DIVINE_IMAGE_EMPOWER,
+            SPELL_PRIEST_DIVINE_IMAGE_EMPOWER_STACK
+        });
+    }
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& eventInfo)
+    {
+        Unit* caster = eventInfo.GetActor();
+        if (!caster)
+            return;
+
+        // Note: if caster has an active Divine Image, we should empower it rather than summoning a new one.
+        if (IsDivineImageSummoned(caster))
+        {
+            Unit* divineImage = ObjectAccessor::GetUnit(*caster, _divineImageGUID);
+            if (!divineImage)
+                return;
+
+            // Note: Divine Image now teleports near the Priest when the Priest casts a Holy Word spell if the Image is further than 15 yards away (Patch 10.1.0).
+            if (caster->GetDistance(divineImage) > 15.0f)
+                divineImage->NearTeleportTo(caster->GetRandomNearPosition(3.0f));
+
+            divineImage->CastSpell(divineImage, SPELL_PRIEST_DIVINE_IMAGE_EMPOWER,
+                CastSpellExtraArgs(aurEff).SetTriggeringSpell(eventInfo.GetProcSpell()));
+        }
+        else
+            caster->CastSpell(*caster, SPELL_PRIEST_DIVINE_IMAGE_SUMMON,
+                CastSpellExtraArgs(aurEff).SetTriggeringSpell(eventInfo.GetProcSpell()));
+
+        caster->CastSpell(caster, SPELL_PRIEST_DIVINE_IMAGE_EMPOWER_STACK,
+            CastSpellExtraArgs(aurEff).SetTriggeringSpell(eventInfo.GetProcSpell()));
+    }
+
+    void Register() override
+    {
+        OnEffectProc += AuraEffectProcFn(spell_pri_divine_image::HandleProc, EFFECT_0, SPELL_AURA_DUMMY);
+    }
+
+public:
+    bool IsDivineImageSummoned(Unit* caster)
```

Comment:
would prefer using something along the lines of `Optional<ObjectGuid> GetDivineImageGUID(Unit* caster)` instead of setting a member variable for this purpose